e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1

```diff
@@ -0,0 +1,477 @@
+    Remove-Item -r .\selfSignedCert 
+}
+
+az keyvault secret download --file .\selfSignedCert --vault-name $ResourceGroup -n $ResourceGroup --encoding base64
+$fileContent = Get-Content .\selfSignedCert -Encoding Byte
+$fileContentB64String = [System.Convert]::ToBase64String($fileContent);
+
+Write-Host "`nSuccessfully fetched the certificate bytes ... removing the cert file from the disk"
+Remove-Item -r .\selfSignedCert
+
+###################################################################################################################################
+# Store all secrets in a KeyVault - Values will be pulled down from here to configure environment variables
+###################################################################################################################################
+
+$SecretsKeyVaultName = "Env$ResourceGroup"
+if($SecretsKeyVaultName.Length -ge 24)
+{
+    $SecretsKeyVaultName = $SecretsKeyVaultName.Substring(0, 24)
+}
+$KeyVaultExists = az keyvault list -g $ResourceGroup --query "[?name=='$SecretsKeyVaultName']" --output tsv
+if($KeyVaultExists -eq $null)
+{
+    Write-Host("`nCreating KeyVault $SecretsKeyVaultName")
+    az keyvault create -g $ResourceGroup --name $SecretsKeyVaultName --output none
+}
+Write-Host("`nWriting secrets to KeyVault $SecretsKeyVaultName")
+az keyvault set-policy -g $ResourceGroup --name $SecretsKeyVaultName --object-id $userObjectId --secret-permissions delete get list set --output none
+az keyvault secret set --vault-name $SecretsKeyVaultName --name "IOTHUB-CONN-STRING-CSHARP" --value $iotHubConnectionString --output none
+az keyvault secret set --vault-name $SecretsKeyVaultName --name "IOTHUB-PFX-X509-THUMBPRINT" --value $iotHubThumbprint --output none
+az keyvault secret set --vault-name $SecretsKeyVaultName --name "IOTHUB-EVENTHUB-CONN-STRING-CSHARP" --value $iotHubConnectionString --output none
+az keyvault secret set --vault-name $SecretsKeyVaultName --name "IOTHUB-EVENTHUB-COMPATIBLE-NAME" --value $eventResourceGroup --output none
+az keyvault secret set --vault-name $SecretsKeyVaultName --name "IOTHUB-EVENTHUB-CONSUMER-GROUP" --value $consumerGroups --output none
+az keyvault secret set --vault-name $SecretsKeyVaultName --name "IOTHUB-PROXY-SERVER-ADDRESS" --value $proxyServerAddress --output none
+az keyvault secret set --vault-name $SecretsKeyVaultName --name "FAR-AWAY-IOTHUB-HOSTNAME" --value "$FarHubName.azure-devices.net" --output none
+az keyvault secret set --vault-name $SecretsKeyVaultName --name "DPS-IDSCOPE" --value $dpsIdScope --output none
+az keyvault secret set --vault-name $SecretsKeyVaultName --name "PROVISIONING-CONNECTION-STRING" --value $deviceProvisioningServiceConnectionString --output none
+az keyvault secret set --vault-name $SecretsKeyVaultName --name "CUSTOM-ALLOCATION-POLICY-WEBHOOK" --value "https://drwe2efnapp.azurewebsites.net/api/HttpTrigger1?code=PtayXB9/P7E5Wt0AmmSs5oO1n0n3jNEM3zDkq3U8VQSziNSMCINOGw==" --output none
```

Comment:
This is referring to my currently deployed function app, including a code.

1. You shouldn't be sharing this publicly.
1. My function app will be removed soon.

Reply:
Good catch! I missed this one. Yes need to get that out too. Will make it a parameter for now where we can send yours and we'll have to figure out how to either automate or create ones for ourselvs.

Reply:
Let's not have people send mine. Perhaps if anything we reuse the PR infra.

+1 for a follow up task to deploy the function app.